### PR TITLE
Add a classes version of iron-flex-layout!

### PR DESCRIPTION
**Pull request by notwaldorf**
Fixes https://github.com/PolymerElements/iron-flex-layout/issues/23 by adding a new classes version of `iron-flex-layout`. Styles are grouped into separated dom-modules so that you don't have to include a giant stylesheet all the time (which could be a performance hit).

Sample use:
```
  <style is="custom-style" include="iron-flex iron-flex-alignment">
   <div class="layout horizontal layout-start">
      <div>cross axis start alignment</div>
   </div>
```

### CODE REVIEW

**Comment by cdata** [comment]
1. Move the `.flex-1` -> `.flex-x` styles into their own file so that we don't maintain them in two places and can take advantage of HTML Imports' natural de-duplication of requests.
 2. "Basic" is the heaviest stylesheet. It seems like there is an opportunity to make "basic" more lightweight by separating the concepts of layout (horizontal, vertical) from item alignment (start, end, justification), and maybe moving item alignment to a more advanced layer of the set (perhaps where `.flex-x` styles belong, or maybe even in its own layer).
 3. These styles don't actually need to be split into multiple files; they just need to be split into multiple dom-modules. Is it better to use multiple files in this case?

**Comment by cdata** [comment]
Also, would it be possible to document somewhere how the layout modules should be consumed to maximize performance outcomes?

**Comment by notwaldorf** [comment]
Yeah, docs would be great. I think that if I combine them all in a giant file, i can also add a giant helpful doc. that won't show up in the catalog because that doesn't work yet :(

**Comment by notwaldorf** [comment]
@cdata oh, #1 is a typo (it's already in its own file, i just double pasted it in `full`. sigh. today is a lot of derping.)

File: classes/iron-flex-layout.html [hunk 1]
```diff
@@ -8,6 +8,10 @@
 subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
 -->
 
+<script>
+  console.warn('This file is deprecated, and will be deleted on October 1st, 2015. Please use `iron-flex-layout/classes/iron-flex-layout-classes.html`, and one of the specific dom-modules instead');
```
Comment (ebidel) [inline classes/iron-flex-layout.html]: back to the future! Can we not specify an exact date? It should be based on a few release versions of Polymer.

Comment (notwaldorf) [inline classes/iron-flex-layout.html]: This PR isn't looking like it's merging any time soon, but I agree -- I'll add more time.

**Comment by nadarik** [comment]
@notwaldorf I had the same idea, but rather than create separate elements I would implement just single element with whole list of classes. Just my humble opinion

**Comment by notwaldorf** [comment]
@nadarik The problem with implementing just a single element (and the reason why we split them up) is that you are bringing in a fairly sizeable CSS file, with a lot of styles that you're not actually going to be using. Since this needs to be done in a lot of elements (because `/deep/` isn't used), it can add up to being quite expensive

**Comment by nadarik** [comment]
@notwaldorf Got it, makes sense to me. Do you know how soon this change gonna be merged?:)

**Comment by notwaldorf** [comment]
I'm going to try to re-jigger it this week!

**Comment by nadarik** [comment]
Thanks a lot

File: classes/iron-flex-layout-classes.html [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+        flex-wrap: wrap;
+      }
+
+      .layout.wrap-reverse {
+        -ms-flex-wrap: wrap-reverse;
+        -webkit-flex-wrap: wrap-reverse;
+        flex-wrap: wrap-reverse;
+      }
+
+      .flex-auto {
```
Comment (sorvell) [inline classes/iron-flex-layout-classes.html]: Suggest having a `.flex` here. Seems even more basic than auto or none =).

File: classes/iron-flex-layout-classes.html [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+        -webkit-align-self: stretch;
+        align-self: stretch;
+      }
+    </style>
+  </template>
+</dom-module>
+
+<dom-module id="iron-flex">
+  <template>
+    <style include="iron-flex-basic"></style>
```
Comment (sorvell) [inline classes/iron-flex-layout-classes.html]: Why is this the only one of these that has a sub-include? Seems like they all should or perhaps none should.

**Comment by sorvell** [comment]
This approach seems fine to me. You might also want to explore including `%` prefixed versions of these modules so that they can be used like this:

```
.my-selector {
  @extends(%layout-horizontal);
}
```

That would be most efficient because the unused % selectors will be eliminated from the css. FYI, we've supported this `@extends` syntax for awhile, but as far as I know it's not heavily used.

**Comment by notwaldorf** [comment]
@sorvell @cdata This is ready for another final look

File: iron-flex-layout-classes.html [hunk 4]
```diff
@@ -0,0 +1,372 @@
...
+      /**
+       * Alignment in main axis.
+       */
+      .layout.start-justified {
+        -ms-flex-pack: start;
+        -webkit-justify-content: flex-start;
+        justify-content: flex-start;
+      }
+
+      .layout.center-justified,
```
Comment (sorvell) [inline iron-flex-layout-classes.html]: This also in the basic one?

File: iron-flex-layout-classes.html [hunk 5]
```diff
@@ -0,0 +1,372 @@
...
+    </style>
+  </template>
+</dom-module>
+
+<dom-module id="iron-flex-factors">
+  <template>
+    <style>
+      .flex,
+      .flex-1 {
+        -ms-flex: 1 1 0.000000001px;
```
Comment (sorvell) [inline iron-flex-layout-classes.html]: Need to figure out if the crazy is still relevant.

File: iron-flex-layout-classes.html (hunk 4 above)
Comment (notwaldorf) [inline iron-flex-layout-classes.html]: I wasn't sure if you would want to use `iron-flex-alignment` _without_ the basic one. You could, I think, in theory, no?

File: iron-flex-layout-classes.html (hunk 5 above)
Comment (notwaldorf) [inline iron-flex-layout-classes.html]: We have tests now. At the moment it's only matching the `iron-flex-layout` mixin. Maybe that should be done in a different PR?

**Comment by notwaldorf** [comment]
@sorvell Answered!

File: iron-flex-layout-classes.html (hunk 4 above)
Comment (sorvell) [inline iron-flex-layout-classes.html]: I thought we decided to put centering into `iron-flex`. Since this is part of centering, can this go there too?

Comment (notwaldorf) [inline iron-flex-layout-classes.html]: Done! Sorry!

**Comment by cdata** [comment]
LGTM :+1:

**Comment by notwaldorf** [comment]
@cdata Yay Yay Yay 🎉 🍻

**Comment by AndreasGalster** [comment]
I have a question about this: Doesn't this just mean we can choose what we can import in our file but we actually still load the **entire** file containing the styles (unless it gets vulcanized)?
